Implement feature section

jbirori opened the pull request:
The following changes are included:

- Implements the new feature section (grid for desktop, scrollview for mobile)
- Include new style file for the apps page
- Adds a new className to the Text component for more specific selecting
- Link to the demo contact form


|     Desktop     |      Mobile      |
| ------------- | ------------- |
|  <img width="1236" alt="Screen Shot 2023-08-18 at 2 33 07 PM" src="https://github.com/DIALUPDIGITAL/portfolio/assets/25678578/ab12ca0f-aa89-47be-9686-004c1ca80901">   |  <img width="404" alt="Screen Shot 2023-08-18 at 2 18 33 PM" src="https://github.com/DIALUPDIGITAL/portfolio/assets/25678578/0a254793-8c5b-4967-a224-4d13f6ce7bc8">  |

jbirori committed:
Implement feature section, include new css file for apps page, and add new className to the Text component

bomanimc commented on src/components/FeatureBlock/FeatureBlock.tsx:
```diff
@@ -0,0 +1,23 @@
+  description,
+}) => (
+  <div className="feature-block">
+    <img src={iconSrc} className="feature-block__img" />
```
Is the BEM style approach more of a personal preference or something that's necessary for namespacing CSS? I just took a look and it seems like the BEM style approach is necessary since I'm not seeing our CSS classes get hashed (lmao rip). I need to go back and edit some of the files I contributed.

jbirori replied:
Just more of a personal preference! Hadn't even noticed the classes weren't getting hashed

bomanimc commented on src/pages/apps.tsx:
```diff
@@ -32,9 +34,40 @@ const IndexPage: React.FC<PageProps> = () => {
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            <Button buttonLabel="request demo" onClick={() => null}/>
```
Hmmm. I wonder if instead of wrapping a button element in a link: should we either just use the raw text here or have a `asLink` prop on the Button component that renders the root element with a link instead of a button. If we go with the normal link approach, maybe we can make a mixin or something that lets us import the button styles to use on other types of elements if needed.

jbirori replied:
Good point! Could also handle the link with JS -> `window.open(url, '_blank').focus()`, any downsides with this?

bomanimc replied:
Down for it. Whatever feels simplest in your opinion!

bomanimc reviewed (approved):
Nice! I feel a bit like there might be a more clean design for the h-scroll section (e.g. maybe using some sort of dot navigation and adding some sort of a border so it doesn't hard-clip the content) or even just having each of the elements in a typical column. Was this more of a first pass on it or something we wanna come back to? I feel like we still have to figure out mobile for this site in general.

jbirori commented:
> Nice! I feel a bit like there might be a more clean design for the h-scroll section (e.g. maybe using some sort of dot navigation and adding some sort of a border so it doesn't hard-clip the content) or even just having each of the elements in a typical column. Was this more of a first pass on it or something we wanna come back to? I feel like we still have to figure out mobile for this site in general.

Yeah this was just an initial pass, we haven't mocked anything out for this section on mobile just yet. With the grid setup the column gets pretty long - down for dot navigation or something adjacent

bomanimc commented:
Sounds good re: design. Feel free to merge whenever!

jbirori committed:
Create global util to demo form in a new tab and add to all the 'request demo' buttons

jbirori commented:
Added a new commit that creates a global util for opening the demo form in a new tab and added that to both of the request demo buttons. Lets look into the alternate mobile interactions for the feature sections in a follow up
